dap-docker.el

```diff
@@ -0,0 +1,253 @@
+;;; dap-docker.el --- Debug Adapter Protocol mode for docker-wrapped servers      -*- lexical-binding: t; -*-
+
+;; Copyright (C) 2022  Andrei Mochalov
+
+;; This program is free software; you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+;; Author: Andrei Mochalov <factyy@gmail.com>
+;; Keywords: languages, debug, docker
+;; URL: https://github.com/emacs-lsp/dap-mode
+;; Package-Requires: ((emacs "26.1") (dash "2.18.0") (lsp-mode "6.0") (f "0.20.0") (s "1.12.0") (ht "2.3") (lsp-docker "1.0.0") (yaml "0.2.0"))
+;; Version: 0.2
+
+;;; Commentary:
+;; Debug Adapter Protocol client support for Emacs and docker-wrapped servers.
+
+;;; Code:
+
+(require 'dap-mode)
+(require 'lsp-mode)
+(require 'lsp-docker)
+(require 'f)
+(require 's)
+(require 'ht)
+(require 'dash)
+(require 'yaml)
+
+(defvar dap-docker-supported-server-types-subtypes
+  (ht ('docker (list 'container 'image)))
+  "A list of all supported server types and subtypes, currently only docker is supported")
+
+(defun dap-docker--is-enabled? (config)
+  "Check whether debugging is enabled"
+  (-if-let* ((server-info (gethash 'debug config))
+             (server-enabled (gethash 'enabled server-info)))
+      (if (equal server-enabled :false)
```

Comment:
nit:

```
(if (equal server-enabled :false)
          nil
        't)
``` 

is equivalent to `(not (equal server-enabled :false))`

Reply:
Fixed in e820d9b